spec/factorial_spec.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../solver'
+
+describe Solver do
+  describe '#factorial' do
+    # Arrange
+    let(:solver) { Solver.new }
+
+    # Act
+    subject { solver.factorial(number) }
+
+    # Assert
+    context 'returns 1 when given 0' do
+      let(:number) { 0 }
+      it { is_expected.to eq(1) }
+    end
+
+    context 'returns 1 when given 1' do
+      let(:number) { 1 }
+      it { is_expected.to eq(1) }
+    end
+
+    context 'returns 2 when given 2' do
+      let(:number) { 2 }
+      it { is_expected.to eq(2) }
+    end
+
+    context 'returns 6 when given 3' do
+      let(:number) { 3 }
+      it { is_expected.to eq(6) }
+    end
+
+    context 'returns 24 when given 4' do
+      let(:number) { 4 }
+      it { is_expected.to eq(24) }
+    end
+
+    context 'returns 120 when given 5' do
+      let(:number) { 5 }
+      it { is_expected.to eq(120) }
+    end
```

Comment:
You did well with your project, kindly note that your factorial should have test that handles negative value which raises an exception as an expectation

![image](https://github.com/Ruthmy/TDD-Practice/assets/20537712/205d1066-b993-4cef-bc8a-21717ed3e5ba)

Reply:
Hi @usorfaitheloho ❗ 
Thank you very much for your feedback, you're totally right, I forgot to add that case. I will add it right away.
Thanks ✨